src/Microsoft.Restier.EntityFramework/Query/QueryExecutor.cs

```diff
@@ -63,16 +53,17 @@ private QueryExecutor()
             IQueryable<TElement> query,
             CancellationToken cancellationToken)
         {
-            long? totalCount = null;
-            if (context.Request.IncludeTotalCount == true)
+#if EF7
+            if (query.Provider is IAsyncQueryProvider)
+#else
+            if (query.Provider is IDbAsyncQueryProvider)
+#endif
             {
-                var countQuery = ExpressionHelpers.GetCountableQuery(query);
-                totalCount = await countQuery.LongCountAsync(cancellationToken);
+                return new QueryResult(
+                    await query.ToArrayAsync(cancellationToken));
             }
 
-            return new QueryResult(
-                await query.ToArrayAsync(cancellationToken),
-                totalCount);
+            return await Inner.ExecuteQueryAsync(context, query, cancellationToken);
```

Comment:
Since DefaultQueryHandler now calls the outermost (WebApi in this case) IQueryExecutor, for EnumerableQuery, EF provider can't handle it correctly, so it must delegate it to inner (DefaultQueryExecutor).